sfdoc/publish/salesforce.py

```diff
@@ -53,34 +69,68 @@ def _get_salesforce_api(self):
             version=settings.SALESFORCE_API_VERSION,
             client_id='sfdoc',
         )
-        return sf
+        cls.api = sf
 
-    def archive(self, ka_id, kav_id):
+    def get_docsets(self):
+        query_str = f"""SELECT Id, {settings.SALESFORCE_DOCSET_ID_FIELD},
+                    Index_Article_Id__c
+                    FROM {settings.SALESFORCE_DOCSET_SOBJECT}"""
+        return self.api.query(query_str)["records"]
+
+    def archive(self, kav_id):
         """Archive a published article."""
+        # Ensure that this article is owned by the right docset
+        article = self.get_by_kav_id(kav_id, "online")
+
+        ka_id = article["KnowledgeArticleId"]
+
+        docset_id = article[self.docset_relation][settings.SALESFORCE_DOCSET_ID_FIELD]
+        assert docset_id == self.docset_uuid
+
+        draft = self.article_info_cache('draft', KnowledgeArticleId=ka_id)
```

Comment:
Since article_info_cache returns a list of potentially multiple results, this would be clearer if you called the variable `drafts`

Reply:
There can only be one draft for a given KA.

Reply:
Yeah but `article_info_cache` returns a (single-item) list, and I expect a list variable to have a plural name. `draft[0]` on the subsequent lines looks funny.

Reply:
Yeah, fair.

Reply:
I think either way is slightly confusing in so I'm going to leave it. Yes, it is a list. But it is a list which contains zero or one item. Which is a slightly confusing concept. Either way the confusion will be cleared up by reading the code right around it. The assertion both validates that Salesforce hasn't gone crazy and helps people to understand why I named it singular.